src/main/java/com/minecolonies/coremod/colony/ColonyManager.java

```diff
@@ -736,28 +771,52 @@ public static boolean backupColonyData()
             saveColonies();
         }
 
-        @NotNull final File file = getSaveLocation();
-        @NotNull final File targetFile = getBackupSaveLocation(new Date());
-        if (!file.exists())
+        try(FileOutputStream fos = new FileOutputStream(getBackupSaveLocation(new Date())))
         {
-            return true;
+            @NotNull final File saveDir = new File(DimensionManager.getWorld(0).getSaveHandler().getWorldDirectory(), FILENAME_MINECOLONIES_PATH);
+            final ZipOutputStream zos = new ZipOutputStream(fos);
+
+            for (int i = 0; i < colonies.size() + BUFFER; i++)
+            {
+                @NotNull final File file = new File(saveDir, String.format(FILENAME_COLONY, i));
+                if (file.exists())
+                {
+                    addToZipFile(String.format(FILENAME_COLONY, i), zos, saveDir);
+                }
+            }
+            addToZipFile(getSaveLocation().getName(), zos, saveDir);
+
+            zos.close();
+            fos.close();
         }
-        else if (targetFile.exists())
+        catch (final Exception e)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)